dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/fileresource/FileResourceCleanUpJob.java

```diff
@@ -67,18 +85,53 @@ public JobType getJobType()
     @Override
     public void execute( JobConfiguration jobConfiguration )
     {
-        List<Pair<String, String>> deleted = new ArrayList<>();
+        FileResourceRetentionStrategy retentionStrategy = (FileResourceRetentionStrategy) systemSettingManager.getSystemSetting(SettingKey.FILE_RESOURCE_RETENTION_STRATEGY);
+
+        List<Pair<String, String>> deletedOrphans = new ArrayList<>();
+
+        List<Pair<String, String>> deletedAuditFiles = new ArrayList<>();
 
         fileResourceService.getOrphanedFileResources()
             .forEach( fr -> {
-                deleted.add( ImmutablePair.of( fr.getName(), fr.getUid() ) );
+                deletedOrphans.add( ImmutablePair.of( fr.getName(), fr.getUid() ) );
                 fileResourceService.deleteFileResource( fr.getUid() );
             } );
 
-        if ( !deleted.isEmpty() )
+        if ( retentionStrategy != FileResourceRetentionStrategy.FOREVER )
         {
-            log.warn( "Deleted " + deleted.size() + " orphaned FileResources: " + prettyPrint( deleted ) );
+            deletedAuditFiles = getExpiredFileResources( retentionStrategy );
+
+            deletedAuditFiles.forEach( pair -> fileResourceService.deleteFileResource( pair.getRight() ) );
         }
+
+        if ( !deletedOrphans.isEmpty() )
+        {
+            log.warn( "Deleted " + deletedOrphans.size() + " orphaned FileResources: " + prettyPrint( deletedOrphans ) );
+        }
+
+        if ( !deletedAuditFiles.isEmpty() )
+        {
+            log.warn( "Deleted " + deletedAuditFiles.size() + " expired FileResource audits: " + prettyPrint( deletedAuditFiles ) );
+        }
+    }
+
+    private List<Pair<String, String>> getExpiredFileResources( FileResourceRetentionStrategy retentionStrategy )
+    {
+        List<Pair<String, String>> expiredFileResources = new ArrayList<>();
+
+        List<DataElement> elements = dataElementService.getAllDataElements().stream()
```

Comment:
Maybe we could add a new method DataElementService.getDataElementsByValueType( ValueType )?